lightly/data/lighty_subset.py

```diff
@@ -0,0 +1,37 @@
+from typing import List, Dict, Tuple
+
+from lightly.data.dataset import LightlyDataset
+
+
+class LightlySubset(LightlyDataset):
+    def __init__(self, base_dataset: LightlyDataset, filenames_subset: List[str]):
+        self.base_dataset = base_dataset
+
+        dict_base_dataset_filename_index: Dict[str, int] = \
+            dict([
```

Comment:
I think we should refrain from using list comprehension when the expressions are more complex. Why not simply use a loop?

Reply:
I assumed that the list comprehension would be faster. However, now that I profiled it, it isn't, thus I changed it to use a for loop.